File: src/admin/modules/sales/service/SalesOrderService.ts
import http from "@/lib/JwtInterceptor";
import { getEnvVariable } from "@/utils/envUtils";

const apiUrl = getEnvVariable('API_URL');

export const SalesOrderService = {
    list: () => {
        return http.get(`${apiUrl}/v1/sales/order/list`);
      },
    
      deleteById: (id: number) => {
        return http.get(`${apiUrl}/v1/sales/order/delete/id/${id}`);
      },
    
      getById: (id: number) => {
        return http.get(`${apiUrl}/v1/sales/order/id/${id}`);
      },
    
      saveOrUpdate: (order: any) => {
        return http.post(`${apiUrl}/v1/sales/order/saveOrUpdate`, order);
      },
    
      printById: (id: number) => {
        return http.get(`${apiUrl}/v1/sales/order/print/${id}`, {
          responseType: 'arraybuffer',
        });
      },
    
      approveById: (id: number) => {
        return http.get(`${apiUrl}/v1/sales/order/approve/id/${id}`);
      },
    
      filter: (pageNumber: number, pageSize: number, search: any) => {
        return http.post(
          `${apiUrl}/v1/sales/order/filter/${pageNumber}/${pageSize}`,
          search
        );
      },
};

export default SalesOrderService;

File: api/src/main/java/com/jee/clinichub/app/sales/order/service/SalesOrderService.java
package com.jee.clinichub.app.sales.order.service;

import java.util.List;

import org.springframework.data.domain.Page;
import com.jee.clinichub.app.sales.order.model.SalesOrderDto;
import com.jee.clinichub.app.sales.order.model.SalesOrderProj;
import com.jee.clinichub.app.sales.order.model.Search;
import com.jee.clinichub.global.model.Status;

public interface SalesOrderService {
	
	SalesOrderDto getById(Long id);

	Status deleteById(Long id);

	Status saveOrUpdate(SalesOrderDto salesOrderDto);

	List<SalesOrderProj> getAllSalesOrders();

	Status approveById(Long id);

	byte[] printById(Long id);

	public 	Page<SalesOrderProj> search(Search search, int pageNo, int pageSize);

}
